test/file-base.js

```diff
@@ -119,3 +119,15 @@ describe('hashSha256', () => {
     )
   })
 })
+
+describe('cached Hash', () => {
+  it('hash is cache to file', async () => {
+    const path_ = 'test/fixtures/sample-cyrillic-encoding.csv'
+    const file = data.open(path_)
+
+    let hash = await file.hash()
+    let cachedHash = await file.hash()
+
+    assert.strictEqual(hash, cachedHash)
+  })
+})
```

Comment:
I think we don't need this. Moreover we don't need to write any new test, well it's out of our scope as I don't think we can test whether its the cache. This test is actually testing whether calling second time the hash is still the same, however if you remove cache implementation this test will still pass.

Reply:
OK, just noticed in the earlier convs @shevron has suggested this test. It's up to you but I think it's extra, or at least it doesn't reflect the requirement for the cache, and also it makes some assumption on the way it is implementation. But it's up to you, just sharing my opinion.

Reply:
Actually, what would you think if the word `cached` was removed from this test? e.g.:

```suggestion
describe('cached Hash', () => {
  it('hash is cache to file', async () => {
    const path_ = 'test/fixtures/sample-cyrillic-encoding.csv'
    const file = data.open(path_)

    let hash1 = await file.hash()
    let hash2 = await file.hash()

    assert.strictEqual(hash1, hash2)
  })
})
```

Now, this just validates a very fundamental functionality of hashing (and of the `File.hash` function) which is that calling `hash` on the same file twice will get the same result. This may sound like testing something trivial, but when you think about it it's actually not, because a lot of things can go wrong when hashing streams, e.g. it is possible that cache is returning a broken hash or that the stream has not been reset before re-calculating the hash or what not. So I think it's a good test, and I agree with your comment that *currently* the naming of variables inside it make assumption about the implementation, but that changing those names will fix that.

Reply:
@shevron agree, it makes sense to me